crates/rust-analyzer/src/lsp_ext.rs

```diff
@@ -98,6 +98,22 @@ pub struct JoinLinesParams {
     pub ranges: Vec<Range>,
 }
 
+pub enum ResolveCodeActionRequest {}
+
+impl Request for ResolveCodeActionRequest {
+    type Params = ResolveCodeActionParams;
+    type Result = Option<SnippetWorkspaceEdit>;
+    const METHOD: &'static str = "experimental/resolveCodeAction";
+}
+
+/// Params for the ResolveCodeActionRequest
+#[derive(Debug, Eq, PartialEq, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ResolveCodeActionParams {
+    pub code_action_params: lsp_types::CodeActionParams,
+    pub label: String,
```

Comment:
Here, we should use `id` and not the `label`. The motivation for this is that (long term) we should be able to select the correct assist by id, without checking even availability. 

Ie, label is a custom text which might be influenced by code at the cursor, while ID is known statically. 

There might be a sligthly complication in that, with group assists, I think there might be two assits with the same ID (ie, two imports for different types). In this case, I think we should keep `id` as a string, but concat an `:<number>` on the LSP layer for diambiguation.